### PR TITLE
fix: Handle each loop

### PR DESCRIPTION
**Pull request by fussybeaver**
This handles an `each` loop around a collection.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -331,6 +332,16 @@ module.exports = grammar({
       ')',
     ),
 
+    each_loop: $ => seq(
+      field('args', $._expression),
```
Comment (fussybeaver) [inline grammar.js]: Perhaps this needs to be more granular... @murtaza64  wdyt?

Comment (murtaza64) [reply grammar.js]: This syntax just seems like a function call. I've tried to implement function calls with optional parentheses but they've given me a huge headache in the past. Does something like
```
collection.each { x -> doSomething(x) }
```
not get parsed properly?

Comment (fussybeaver) [reply grammar.js]: happy to have it be a function call. it errored on the dot operator after a list... having some PTO atm but happy with what you come up with or i can take another stab at in early September.

Comment (murtaza64) [reply grammar.js]: fair enough. I think it's better if we can get 
```
["list", "literal"].anyMethod()
```
working. I think we just need to update `_prefix_expression` to include list lierals (https://github.com/murtaza64/tree-sitter-groovy/blob/main/grammar.js#L126-L132). Hopefully it's as easy as that and doesn't cause any conflicts

currently don't have time to look into this unfortunately

Comment (fussybeaver) [reply grammar.js]: Ok, I've just added `list` to `_prefix_expression` and reworked the test case... Hopefully that should be enough.

**Pull request merged**